src/NUnitFramework/framework/Api/FrameworkController.cs

```diff
@@ -174,7 +177,15 @@ private void Initialize(string assemblyPath, IDictionary settings)
         /// <summary>
         /// Gets a dictionary of settings for the FrameworkController
         /// </summary>
-        public IDictionary Settings { get; private set; }
+        public IDictionary Settings
+        {
+            get { return (IDictionary)InternalSettings; }
+        }
+
+        /// <summary>
+        /// Gets a dictionary of settings for the FrameworkController
+        /// </summary>
+        internal IDictionary<string, object> InternalSettings { get; private set; }
```

Comment:
I don't see the need and frankly would have prefered to review without that change, as I originally suggested. We seem to be fixing a compatibility problem that @oznetmaster sees with the framework API, but I don't see this property as a part of that API. The API description indicates that the names and accepted types of the constructors for the FrameworkController and it's subtask classes must remain constant. That's all.

Since users of the API never reference the controller object as anything other than object, the Settings property is not available to them. We should check to see if it's referenced by anything other than tests and, if not, change its visibility.

I would prefer to see the whole business with InternalSettings dropped as an unnecessary complication.